src/Contacts.ts

```diff
@@ -0,0 +1,205 @@
+import { Envelope } from '@xmtp/proto/ts/dist/types/message_api/v1/message_api.pb'
+import Client from './Client'
+import {
+  // eslint-disable-next-line camelcase
+  ecies_decrypt_k256_sha3_256,
+  // eslint-disable-next-line camelcase
+  ecies_encrypt_k256_sha3_256,
+  // eslint-disable-next-line camelcase
+  generate_private_preferences_topic,
+} from '@xmtp/ecies-bindings-wasm'
+import { privatePreferences } from '@xmtp/proto'
+import { buildUserPrivatePreferencesTopic } from './utils'
+import { PublishParams } from './ApiClient'
+
+export type AllowListPermissionType = 'allow' | 'block' | 'unknown'
+
+export type AllowListEntryType = 'address'
+
+export class AllowListEntry {
+  value: string
+  entryType: AllowListEntryType
+  permissionType: AllowListPermissionType
+
+  constructor(
+    value: string,
+    entryType: AllowListEntryType,
+    permissionType: AllowListPermissionType
+  ) {
+    this.value = value
+    this.entryType = entryType
+    this.permissionType = permissionType
+  }
+
+  get key(): string {
+    return `${this.entryType}-${this.value}`
+  }
+
+  static fromAddress(
+    address: string,
+    permissionType: AllowListPermissionType = 'unknown'
+  ): AllowListEntry {
+    return new AllowListEntry(address, 'address', permissionType)
+  }
+}
+
+export class AllowList {
+  entries: Map<string, AllowListPermissionType>
+
+  constructor() {
+    this.entries = new Map<string, AllowListPermissionType>()
+  }
+
+  allow(address: string) {
+    const entry = AllowListEntry.fromAddress(address, 'allow')
+    this.entries.set(address, 'allow')
+    return entry
+  }
+
+  block(address: string) {
+    const entry = AllowListEntry.fromAddress(address, 'block')
+    this.entries.set(address, 'block')
+    return entry
+  }
+
+  state(address: string) {
+    return this.entries.get(address) ?? 'unknown'
+  }
+
+  static async load(client: Client): Promise<AllowList> {
+    const allowList = new AllowList()
+    const privateKeyBundle = await client.keystore.getPrivateKeyBundle()
```

Comment:
This will break in the browser when using Snaps, since this method isn't supported in a Snap.

You're actually going to have to add the decryption methods to the Keystore and follow the Keystore release process [outlined here](https://github.com/xmtp/xmtp-js/tree/main/src/keystore).

Apologies for not warning you about that wrinkle. We can test in the `beta` branch while we wait for MetaMask to support our new Snap release, but can't release this version of the SDK until there is a corresponding Snap version.